public/js/controllers/charts_controller.js

```diff
@@ -197,20 +222,11 @@ export default class extends Controller {
     baseSubsidy = parseInt(this.data.get('bs'))
     subsidyInterval = parseInt(this.data.get('sri'))
     subsidyExponent = parseFloat(this.data.get('mulSubsidy')) / parseFloat(this.data.get('divSubsidy'))
+    windowSize = parseInt(this.data.get('windowSize'))
 
-    this.settings = TurboQuery.nullTemplate(['chart', 'zoom', 'scale', 'bin'])
-    this.query.update(this.settings)
-    if (this.settings.zoom) {
-      this.setSelectedZoom(this.settings.zoom)
```

Comment:
See notes below about why you can't replace `setSelectedZoom` with `setZoom` quite so easily.